pkg/tree/filetree.go

```diff
@@ -154,6 +164,35 @@ func (t *FileTree) AddPath(path file.Path) (file.Reference, error) {
 	return f, nil
 }
 
+// AddPathAndMissingAncestors adds a new path to the tree. It also adds any
+// ancestors of the path that are not already present in the tree. The resulting
+// file.Reference of the // new (leaf) addition is returned.
+func (t *FileTree) AddPathAndMissingAncestors(path file.Path) (file.Reference, error) {
```

Comment:
nit: maybe consider `AddPathAndAncestors` or `AddPathWithAncestors` instead? The "Missing" is somewhat redundant since `AddPath` only adds paths that have not already been added, so in similar fashion adding a path and ancestors doesn't need an indication of missing-or-not in the name. (Also makes the name shorter without losing meaning, which is an API win)